src/main/scala/com/spark3d/spatial3DRDD/Shape3DRDD.scala

```diff
@@ -111,10 +111,15 @@ abstract class Shape3DRDD[T<:Shape3D] extends Serializable {
         new OnionPartitioner(grids)
       }
       case GridType.OCTREE => {
-        val octree = new Octree(getDataEnvelope, 0)
         // taking 20% of the data as a sample
         val sampleSize = (rawRDD.count * 0.2).asInstanceOf[Int]
         val samples = rawRDD.takeSample(false, sampleSize, 12).toList.map(x => x.getEnvelope)
+        // TO DO
```

Comment:
The fact that we have only 1 partition is not specific to spark-fits, but rather due to the size of the test data you are loading. By default, spark-fits (or any Spark connector, including CSV) will shape the data in partitions of size ~128 MB (or whatever size is defined for the blocks of the underlying HDFS). I do not think the initial partitioning matters much (expect for the reshuffling) as spark3D will redefine the number of partitions needed anyway.

Reply:
Ok makes sense. So even the spark-fits adheres to the standard HDFS block size of 64MB. No need to refactor. Will remove the comment.